test/integration/integration/tests/resources/tests/custom-policy-ratelimiting.yaml

```diff
@@ -0,0 +1,112 @@
+# --------------------------------------------------------------------
+# Copyright (c) 2023, WSO2 LLC. (http://wso2.com) All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------
+apiVersion: dp.wso2.com/v1alpha1
+kind: RateLimitPolicy
+metadata:
+  name: http-bin-ratelimit-usergroup
+  namespace: gateway-integration-test-infra
+spec:
+  override:
+    type: Custom
+    custom:
+      key: rlkey_user
+      value: bob
+      rateLimit:
+        requestsPerUnit: 4
+        unit: Minute
+    organization: default
+  targetRef:
+    kind: Gateway
+    name: default
+    group: gateway.networking.k8s.io
+---
+apiVersion: dp.wso2.com/v1alpha1
+kind: APIPolicy
+metadata:
+  name: interceptor-api-policy-api-level
+  namespace: gateway-integration-test-infra
+spec:
+  override:
+    requestInterceptor:
+      backendRef:
+        name: interceptor-backend-v1
+      includes:
+        - request_body
+        - request_headers
+        - invocation_context
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: prod-http-route-http-bin-api-basic
```

Comment:
did you test this on top of current main branch code? this kind is not supported now. it should be API or Resource.

Reply:
Tested yesterday on the main but I think it was before #963 got merged. Will change the CRs to the new format.

Reply:
Updated the test CRs after testing the feature locally with #963 changes.